test/spec/Menu-test.js

```diff
@@ -734,6 +734,35 @@ define(function (require, exports, module) {
                     expect(menu).toBeTruthy();   // Verify that we got this far...
                 });
             });
+
+            it("should add then remove new menu item ensuring event listeners have also been detached", function () {
+                runs(function () {
+                    var menuItemId = "menu-test-removeMenuItem4";
+                    var commandId = "Menu-test.removeMenuItem.command4";
+                    CommandManager.register("Brackets Test Command Custom", commandId, function () {});
+                    var menu = Menus.addMenu("Custom", menuItemId);
+
+
+                    var menuItem = menu.addMenuItem(commandId);
+
+                    var command = CommandManager.get(commandId);
```

Comment:
to be sure you're detaching only stuff that's been added by `addMenuItem` I'd suggest adding a custom event here like this: `command.on("nameChange", function () {})` and that handler should stay attached even after you call `removeMenuItem`